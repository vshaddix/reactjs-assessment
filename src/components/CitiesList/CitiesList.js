import React from "react";
import CityCard from '../CityCard/CityCard';

export default ({ cities, count = 20 }) => {
  if (!cities) {
    return <div class="ui loader" />;
  }

  const items = cities.slice(0, count).map(city => {
    return (
      <CityCard
        key={city.city + city.country}
        city={city.city}
        country={city.country}
      />
    );
  });


  return (
    <div className="ui cards">
      {items}
    </div>
  );
};
