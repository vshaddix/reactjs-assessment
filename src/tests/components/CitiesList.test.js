import React from 'react';
import { create } from "react-test-renderer";
import CitiesList from '../../components/CitiesList/CitiesList';

jest.mock('../../services/MeasurementsService', () => {
  return function () {
    return {
      getMeasurementsForACity: () => {
        return new Promise(resolve => resolve([]))
      }
    }
  };
});

it('renders (snapshot test)', () => {
  const component = create(<CitiesList />);

  expect(component.toJSON()).toMatchSnapshot();
});

it('renders correctly when at least one object with city and country are passed', () => {
  const cities = [
    {
      city: "Sofiq",
      country: "Bulgaria"
    },
    {
      city: "Plovediv",
      country: "Bulgaria"
    }
  ];
  const parametersToDisplay = ['pm25'];
  const component = create(<CitiesList cities={cities} parametersToDisplay={parametersToDisplay} />);

  expect(component.toJSON()).toMatchSnapshot();
});

it('renders correctly when a count is passed', () => {
  const cities = [
    {
      city: "Sofiq",
      country: "Bulgaria"
    },
    {
      city: "Plovediv",
      country: "Bulgaria"
    }
  ];
  const parametersToDisplay = ['pm25'];
  const component = create(<CitiesList cities={cities} parametersToDisplay={parametersToDisplay} count={1}/>);

  expect(component.toJSON()).toMatchSnapshot();
});
