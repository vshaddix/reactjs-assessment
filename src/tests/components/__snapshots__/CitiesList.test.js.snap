// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders (snapshot test) 1`] = `
<div
  class="ui loader"
/>
`;

exports[`renders correctly when a count is passed 1`] = `
<div
  className="ui cards"
>
  <div
    className="ui card"
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Sofiq
      </div>
    </div>
    <div
      className="content"
    >
      <h4
        className="ui sub header"
      >
        Measurements
      </h4>
      <div
        className="ui placeholder"
      >
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly when at least one object with city and country are passed 1`] = `
<div
  className="ui cards"
>
  <div
    className="ui card"
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Sofiq
      </div>
    </div>
    <div
      className="content"
    >
      <h4
        className="ui sub header"
      >
        Measurements
      </h4>
      <div
        className="ui placeholder"
      >
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
      </div>
    </div>
  </div>
  <div
    className="ui card"
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Plovediv
      </div>
    </div>
    <div
      className="content"
    >
      <h4
        className="ui sub header"
      >
        Measurements
      </h4>
      <div
        className="ui placeholder"
      >
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
        <div
          className="line"
        />
      </div>
    </div>
  </div>
</div>
`;
